### DISCUSSION TITLE
ENV Variable for ValheimPlus

### DISCUSSION THREAD

**Discussion by stephenbock** [mbround18/valheim-docker#690] [Ideas]
With the latest Valheim patch, Valheim Plus is no longer working from the main repo.  There is a [community fix](https://github.com/Grantapher/ValheimPlus) out there, but that lives in a different repo.  It would be great if we could specify the Valheim Plus repo as an env variable so that we don't have to manually copy files every time we build a new container.

**Comment by mbround18** [comment]
Great feature request! Ill take a look at it!

**Comment by mtannertdev** [comment]
I want to add to that this is key. I had to do just this last night and I did it by binding/mounting a 'tweaked' version of 'start_valheim.sh' to the docker container to use a different URL. I don't think that's something many people are going to do. This is going to be a necessary feature going forward as who knows where the next Valheim+ update comes from.

**Reply by mbround18** [reply]
@mtannertdev  i could do a VALHEIM_PLUS_DOWNLOAD_URL variable, thoughs?

**Reply by mbround18** [reply]
https://github.com/mbround18/valheim-docker/pull/696

**Reply by mtannertdev** [reply]
In all honesty, that might be a bit of overkill. There might be some people who want to granularly control the version of bepinex, but I think all that is needed by the community to support the current alt-address of Valheim+ would be something like this pseudo code in the start_valheim:

```
install_valheim_plus() {
    log "Installing ValheimPlus"
    if VALHEIM_PLUS_DOWNLOAD_URL is an env variable set? then...
      use the provided "${VALHEIM_PLUS_DOWNLOAD_URL
    else if  VALHEIM_PLUS_RELEASES_URL is an env variable set? then...
      use VALHEIM_PLUS_RELEASES_URL to get VALHEIM_PLUS_DOWNLOAD_URL
    else
       use default hardcoded VALHEIM_PLUS_RELEASES_URL  to get VALHEIM_PLUS_DOWNLOAD_URL
    fi
    log "Pulling ValheimPlus from ${VALHEIM_PLUS_DOWNLOAD_URL}"
    odin mod:install "${VALHEIM_PLUS_DOWNLOAD_URL}"
}
```

The above psuedo code would allow two options: either a specific direct URL to the file you want to download, or an ability to grab a latest release from an alternative github repo (Which is what some people are doing now for the patched valheim by Grant).
